lib/core/src/Cardano/Wallet/DB/Store/Wallets/Store.hs

```diff
@@ -17,109 +19,155 @@ Implementation of a store for 'TxWalletsHistory'
+        wids <- lift $ fmap (view #txMetaWalletId . entityVal)
             <$> selectList @TxMeta [] []
-        runExceptT $ do
-            xs <- forM wids $ ExceptT . loadS . mkStoreMetaTransactions
-            pure $ Map.fromList $ zip wids xs
+        subsWids <- lift $ fmap (view #localTxSubmissionWalletId . entityVal)
+            <$> selectList @LocalTxSubmission [] []
+        fmap Map.fromList
+            $ forM (nub $ wids <> subsWids) $ \wid -> (wid,)
+                <$> ExceptT (loadS $ mkStoreWalletMetaWithSubmissions wid)
 
--- | Store for 'DeltaTxWalletsHistory'.
-mkStoreTxWalletsHistory
-    :: Store (SqlPersistT IO) DeltaTxWalletsHistory
+
+mkStoreTxWalletsHistory :: Store (SqlPersistT IO) DeltaTxWalletsHistory
 mkStoreTxWalletsHistory =
     Store
     { loadS =
-            liftA2 (,) <$> loadS mkStoreTransactions <*> loadS mkStoreWalletsMeta
+          liftA2 (,)
+            <$> loadS mkStoreTransactions
+            <*> loadS mkStoreWalletsMetaWithSubmissions
     , writeS = \(txHistory,txMetaHistory) -> do
-            writeS mkStoreTransactions txHistory
-            writeS mkStoreWalletsMeta txMetaHistory
+          writeS mkStoreTransactions txHistory
+          writeS mkStoreWalletsMetaWithSubmissions txMetaHistory
     , updateS = \(txh@(TxHistoryF mtxh),mtxmh) -> \case
-            ExpandTxWalletsHistory wid cs -> do
-                updateS mkStoreTransactions txh
-                    $ TxStore.Append
-                    $ mkTxHistory
-                    $ fst <$> cs
-                -- see also Store.Wallets.Model.garbageCollectEmptyWallets
-                updateS mkStoreWalletsMeta mtxmh
-                    $ case Map.lookup wid mtxmh of
-                        Nothing ->
-                            Insert wid $ mkTxMetaHistory wid cs
-                        Just _ ->
-                            Adjust wid
-                            $ TxMetaStore.Expand
-                            $ mkTxMetaHistory wid cs
             ChangeTxMetaWalletsHistory wid change
-                -> updateS mkStoreWalletsMeta mtxmh
-                $ Adjust wid
-                $ Manipulate change
+                -> updateS mkStoreWalletsMetaWithSubmissions mtxmh
+                $ Adjust wid change
             GarbageCollectTxWalletsHistory -> mapM_
                 (updateS mkStoreTransactions txh . DeleteTx)
                 $ Map.keys
                 $ Map.withoutKeys mtxh
                 $ walletsLinkedTransactions mtxmh
-            RemoveWallet wid -> updateS mkStoreWalletsMeta mtxmh $ Delete wid
+            RemoveWallet wid -> updateS mkStoreWalletsMetaWithSubmissions mtxmh
+                $ Delete wid
+            ExpandTxWalletsHistory wid cs -> do
+                updateS mkStoreTransactions txh
+                    $ TxStore.Append
+                    $ mkTxHistory
+                    $ fst <$> cs
+                updateS mkStoreWalletsMetaWithSubmissions mtxmh
+                    $ Adjust wid
```

Comment:
We have to make a case distinction and apply `Insert` when the map is empty, see other comments.